File: kanban-service/app/models/workspace.py
from sqlalchemy import Column, String, Text
from app.models.base import BaseModel

class Workspace(BaseModel):
    __tablename__ = "workspaces"

    name = Column(String(100), nullable=False, index=True)
    description = Column(Text, nullable=True)

    def __repr__(self):
        return f"<Workspace(id={self.id}, name={self.name})>"

File: board-service/internal/domain/workspace.go
package domain

import "github.com/google/uuid"

type Workspace struct {
	BaseModel
	Name        string    `gorm:"type:varchar(255);not null" json:"name"`
	Description string    `gorm:"type:text" json:"description"`
	OwnerID     uuid.UUID `gorm:"type:uuid;not null" json:"owner_id"`
	IsPublic    bool      `gorm:"default:false" json:"is_public"`
}

func (Workspace) TableName() string {
	return "workspaces"
}
